app/serializers/event_serializer.rb

```diff
@@ -11,7 +11,7 @@ class EventSerializer < ActiveModel::Serializer
   has_one :vote, serializer: VoteSerializer
 
   def actor
-    object.eventable.try(:user)
+    object.user || object.eventable.try(:user)
```

Comment:
@gdpelican, @robguthrie and I are gonna have a stab at refactoring this whole system tomorrow morning, so stay tuned!